samplers/aws/xray/internal/client_test.go

```diff
@@ -261,3 +261,22 @@ func TestEndpointIsNotReachable(t *testing.T) {
 	_, err = client.getSamplingRules(context.Background())
 	assert.Error(t, err)
 }
+
+func TestRespondsWithErrorStatusCode(t *testing.T) {
+	testServer := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, _ *http.Request) {
+		res.WriteHeader(http.StatusForbidden)
+		_, err := res.Write([]byte(`{}`))
+		require.NoError(t, err)
+	}))
+	t.Cleanup(testServer.Close)
+
+	u, err := url.Parse(testServer.URL)
+	require.NoError(t, err)
+
+	client, err := newClient(*u)
+	require.NoError(t, err)
+
+	samplingRules, err := client.getSamplingRules(context.Background())
+	require.Error(t, err)
```

Comment:
Could you add an assertions checking that the error content matches the proper error? So if somehow, another error is returned, this doesn't still pass.

Reply:
Sure, no problem.
I tried to adapt to the existing tests where errors are not explicitly asserted. 

I would recommend to add more specific assertions on all tests related to errors or even add new ones to distinguish error scenarios. 

If that’s fine for you I would work on that.